thorlcr/master/thmastermain.cpp

```diff
@@ -840,7 +840,16 @@ int main( int argc, char *argv[]  )
         masterSlaveMpTag = allocateClusterMPTag();
         kjServiceMpTag = allocateClusterMPTag();
 
-        if (registry->connect())
+        if (0 == numSlaves) // bare metal
+        {
+            unsigned localThorPortInc = globals->getPropInt("@localThorPortInc", DEFAULT_SLAVEPORTINC);
+            unsigned slaveBasePort = globals->getPropInt("@slaveport", DEFAULT_THORSLAVEPORT);
+            Owned<IGroup> rawGroup = getClusterNodeGroup(thorname, "ThorCluster");
+            setClusterGroup(queryMyNode(), rawGroup, slavesPerNode, channelsPerSlave, slaveBasePort, localThorPortInc);
```

Comment:
Does this need to be called at some point if numSlaves != 0? (E.g. after connection?)  I think it is superseded by the call to setupCluster(), but a comment might make that clearer

Reply:
yes they (setupCluster and setupClusterGroup) are essentially doing the same thing, with different assumptions.
I'll add a comment.

Reply:
in fact in master after  setClusterGroup, setupCluster didn't need calling (it only duplicated what had already been don).
Have changed it so no longer called.